### PR TITLE
*: rebase controller runtime refactor onto master and add ao commands back

### PR DESCRIPTION
**Pull request by shawn-hurley**
/cc @johnkim76 and @dymurray can you guys help me make sure the ansible operator commands are correct.

### CODE REVIEW

File: commands/operator-sdk/cmd/test/local.go [hunk 1]
```diff
@@ -22,7 +22,9 @@ import (
 	"os/exec"
 	"strings"
```
Comment (estroz) [inline commands/operator-sdk/cmd/test/local.go]: Remove newline.

File: commands/operator-sdk/cmd/new.go [hunk 2]
```diff
@@ -143,50 +231,56 @@ func repoPath() string {
...
+		if kindFirstLetter != strings.ToUpper(kindFirstLetter) {
+			log.Fatal("--kind must start with an uppercase letter")
+		}
+		if strings.Count(apiVersion, "/") != 1 {
+			log.Fatalf("api-version has wrong format (%v); format must be $GROUP_NAME/$VERSION (e.g app.example.com/v1alpha1)", apiVersion)
+		}
 	}
 }
 
 func mustGetwd() string {
```
Comment (estroz) [inline commands/operator-sdk/cmd/new.go]: Remove.

File: pkg/scaffold/ansible/dockerfile.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+type Dockerfile struct {
+	input.Input
+
+	GeneratePlaybook bool
+}
+
+// GetInput - gets the input
+func (d *Dockerfile) GetInput() (input.Input, error) {
+	if d.Path == "" {
+		d.Path = filepath.Join("build", "Dockerfile")
```
Comment (estroz) [inline pkg/scaffold/ansible/dockerfile.go]: Use constants in `constants.go`, and add new constants as needed.

File: pkg/scaffold/ansible/galaxy_init.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	Kind string
+	Dir  string
+}
+
+// GetInput - get input
+func (g *GalaxyInit) GetInput() (input.Input, error) {
+	if g.Path == "" {
+		dir, err := ioutil.TempDir("", "osdk")
+		if err != nil {
+			return g.Input, err
```
Comment (estroz) [inline pkg/scaffold/ansible/galaxy_init.go]: `return Input{}, err`

File: pkg/scaffold/ansible/galaxy_init.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+	"path/filepath"
+
+	"github.com/operator-framework/operator-sdk/pkg/scaffold/input"
+)
+
+// GalaxyInit - wrapper
+type GalaxyInit struct {
+	input.Input
+
+	Kind string
```
Comment (estroz) [inline pkg/scaffold/ansible/galaxy_init.go]: This should have a full `Resource` field, which should be set in `cmd/new.go`.

File: pkg/scaffold/ansible/galaxy_init.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+	Dir  string
+}
+
+// GetInput - get input
+func (g *GalaxyInit) GetInput() (input.Input, error) {
+	if g.Path == "" {
+		dir, err := ioutil.TempDir("", "osdk")
```
Comment (estroz) [inline pkg/scaffold/ansible/galaxy_init.go]: Temp dir creation should be within project context, in case scaffolding fails.

File: pkg/scaffold/ansible/galaxy_init.go [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+		if err != nil {
+			return g.Input, err
+		}
+		g.Path = filepath.Join(dir, "galaxy_init.sh")
```
Comment (estroz) [inline pkg/scaffold/ansible/galaxy_init.go]: Move filename to `constants.go`.

File: pkg/scaffold/ansible/playbook.go [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+type Playbook struct {
+	input.Input
+
+	Kind string
+}
+
+// GetInput - gets the input
+func (p *Playbook) GetInput() (input.Input, error) {
+	if p.Path == "" {
+		p.Path = "playbook.yaml"
```
Comment (estroz) [inline pkg/scaffold/ansible/playbook.go]: Move filename to `constants.go`.

File: pkg/scaffold/ansible/playbook.go [hunk 9]
```diff
@@ -0,0 +1,26 @@
+package ansible
+
+import "github.com/operator-framework/operator-sdk/pkg/scaffold/input"
+
+// Playbook - the playbook tmpl wrapper
+type Playbook struct {
+	input.Input
+
+	Kind string
```
Comment (estroz) [inline pkg/scaffold/ansible/playbook.go]: This should have a full `Resource` field, which should be set in `cmd/new.go`.

File: pkg/scaffold/ansible/watches_yaml.go [hunk 10]
```diff
@@ -0,0 +1,34 @@
+package ansible
+
+import (
+	"github.com/operator-framework/operator-sdk/pkg/scaffold"
+	"github.com/operator-framework/operator-sdk/pkg/scaffold/input"
+)
+
+const (
+	watchesFile = "watches.yaml"
```
Comment (estroz) [inline pkg/scaffold/ansible/watches_yaml.go]: Move to `constants.go`

File: pkg/scaffold/gitignore.go [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+.*.sw[a-p]
+# session
+Session.vim
+# temporary
+.netrwhist
+# auto-generated tag files
+tags
+### VisualStudioCode ###
+.vscode/*
+!.vscode/settings.json
```
Comment (estroz) [inline pkg/scaffold/gitignore.go]: Remove these regexp's if you're removing them from the SDK's `.gitignore`.

File: commands/operator-sdk/cmd/new.go [hunk 12]
```diff
@@ -76,50 +73,141 @@ const (
...
+		}
+	}
+
+	// Run galaxy init.
+	cmd := exec.Command(filepath.Join(galaxyInit.AbsProjectPath, galaxyInit.Path))
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.Run()
+	// Delete Galxy INIT
+	err = os.RemoveAll(filepath.Join(galaxyInit.AbsProjectPath, "var"))
```
Comment (dymurray) [inline commands/operator-sdk/cmd/new.go]: This isn't working for me:
```
$ operator-sdk new --type ansible foo-operator --api-version foo.example.com/v1alpha1 --kind FOo
$ ls foo-operator/tmp/osdk494667550/
galaxy_init.sh
```
```

File: pkg/scaffold/operator.go [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+  replicas: 1
+  selector:
+    matchLabels:
+      name: {{.ProjectName}}
+  template:
+    metadata:
+      labels:
+        name: {{.ProjectName}}
+    spec:
+      containers:
```
Comment (estroz) [inline pkg/scaffold/operator.go]: There are Ansible template pipelines in [`master`](https://github.com/operator-framework/operator-sdk/pull/486/files#diff-e35d7b9b13729eab0761345f9ffbf202R459) that aren't here. Are they still required?

File: pkg/scaffold/ansible/galaxy_init.go (hunk 4 above)
Comment (shawn-hurley) [reply pkg/scaffold/ansible/galaxy_init.go]: Why is the empty input preferable to the g.Input?

File: pkg/scaffold/ansible/galaxy_init.go (hunk 6 above)
Comment (estroz) [inline pkg/scaffold/ansible/galaxy_init.go]: Most state derived at runtime should not be set in `GetInput()`. Move tmp dir creation and `g.Dir` setting responsibility to the caller.

File: commands/operator-sdk/cmd/new.go (hunk 12 above)
Comment (dymurray) [reply commands/operator-sdk/cmd/new.go]: Need to change `var` to `os.TempDir()`

File: commands/operator-sdk/cmd/new.go [hunk 14]
```diff
@@ -76,50 +73,141 @@ const (
 	ensureCmd           = "ensure"
 	goOperatorType      = "go"
 	ansibleOperatorType = "ansible"
+
```
[the remark this answers is not in the document]
Comment (shawn-hurley) [reply commands/operator-sdk/cmd/new.go]: They shouldn't because we don't want our users to have to input 0 for go and 1 for Ansible. I think they probably need to stay as they are.

File: Makefile [hunk 15]
```diff
@@ -9,7 +9,7 @@ else
        Q = @
 endif
 
-VERSION = $(shell git describe --dirty --tags)
+VERSION = $(shell git describe --dirty --tags || echo UNKNOWN)
```
Comment (estroz) [inline Makefile]: Why was this added?

File: commands/operator-sdk/cmd/new.go (hunk 14 above)
[the remark this answers is not in the document]
Comment (estroz) [reply commands/operator-sdk/cmd/new.go]: My comment refers to the `default...Mode` variables below.

File: pkg/scaffold/ansible/galaxy_init.go (hunk 4 above)
Comment (estroz) [reply pkg/scaffold/ansible/galaxy_init.go]: When returning an error you shouldn't return other computed data. `g.Input` fields might have been set by a previous call to `GetInput()` or modified elsewhere. Returning an empty `Input{}` is cleaner.

File: Makefile (hunk 15 above)
Comment (shawn-hurley) [reply Makefile]: git Shallow copy can have no tags and this will fail. if it fails we add UNKNOWN so that travis is not broken here. at least that is my understanding

Comment (mhrivnak) [reply Makefile]: Travis does a shallow checkout with a depth of 50. If none of the commits it retrieves have a tag, this `git describe` command will fail. We were hitting that problem this morning.

Comment (AlexNPavel) [reply Makefile]: The `make install` command started failing for this branch since travis by default uses a 50 commit depth in its pull. The last tag was more than 50 commits ago, so `make install` would not work correctly. This basically just makes it so that if no tag is found, we just use `UNKNOWN`

Comment (mhrivnak) [reply Makefile]: lol that we all answered this at the same time.

Comment (estroz) [reply Makefile]: :+1:

File: commands/operator-sdk/cmd/new.go [hunk 16]
```diff
@@ -203,3 +310,66 @@ func initGit() {
...
+	roleYAML, err := ioutil.ReadFile(roleFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read role manifest %v: %v", roleFilePath, err)
+	}
+	obj, _, err := cgoscheme.Codecs.UniversalDeserializer().Decode(roleYAML, nil, nil)
+	if err != nil {
+		return fmt.Errorf("failed to decode role manifest %v: %v", roleFilePath, err)
+	}
+	switch role := obj.(type) {
+	// TODO: use rbac/v1.
```
Comment (AlexNPavel) [inline commands/operator-sdk/cmd/new.go]: This TODO is completed

Comment (hasbro17) [reply commands/operator-sdk/cmd/new.go]: Shawn I know you were thinking about the possibility of generating and handling ClusterRoles in the operator(along with watching all namesapces).
For now we update the existing role.yaml with a new rule whenever add a new API `operator-sdk add api`.
```Go
case *rbacv1.Role:
...
default:
	return errors.New("failed to parse role.yaml as a role")
```
Can you change this TODO to handle ClusterRole.

File: test/e2e/memcached_test.go [hunk 17]
```diff
@@ -118,7 +142,7 @@ func TestMemcached(t *testing.T) {
 				t.Fatal(err)
 			}
 			// TODO: make this match more complete in case we add another repo tracking master
-			gopkg = bytes.Replace(gopkg, []byte("branch = \"master\""), []byte("# branch = \"master\""), -1)
+			gopkg = bytes.Replace(gopkg, []byte("branch = \"refactor/controller-runtime\""), []byte("# branch = \"refactor/controller-runtime\""), -1)
```
Comment (AlexNPavel) [inline test/e2e/memcached_test.go]: This should be matched to the branch that we are currently on. These branches are close enough that they are not breaking, but we should probably switch this back to master once we merge this PR and then figure out how to handle this best (this will become a bit more of a problem when we start having release branches).

Comment (AlexNPavel) [reply test/e2e/memcached_test.go]: This actually is a change that has to happen in 2 places: here and in the gopkg toml scaffold. Not changing this shouldn't break the PR tests since we're just commenting out the branch, but it would break the master branch and users if not changed.

File: README.md [hunk 18]
```diff
@@ -58,37 +58,48 @@ Create and deploy an app-operator using the SDK CLI:
...
 # Cleanup
-$ kubectl delete -f deploy/cr.yaml
-$ kubectl delete -f deploy/crd.yaml
+$ kubectl delete -f deploy/app_v1alpha1_appservice_cr.yaml
 $ kubectl delete -f deploy/operator.yaml
-$ kubectl delete -f deploy/rbac.yaml
-$ kubectl delete -f deploy/sa.yaml
+$ kubectl delete -f deploy/role.yaml
+$ kubectl delete -f deploy/role_binding.yaml
+$ kubectl delete -f deploy/app_v1alpha1_appservice_crd.yaml
```
Comment (LiliC) [inline README.md]: This should be `kubectl delete -f  deploy/crds/app_v1alpha1_appservice_crd.yaml `

File: README.md [hunk 19]
```diff
@@ -58,37 +58,48 @@ Create and deploy an app-operator using the SDK CLI:
...
 # Create an app-operator project that defines the App CR.
 $ mkdir -p $GOPATH/src/github.com/example-inc/
+# Create a new app-operator project
 $ cd $GOPATH/src/github.com/example-inc/
-$ operator-sdk new app-operator --api-version=app.example.com/v1alpha1 --kind=App
+$ operator-sdk new app-operator
 $ cd app-operator
 
+# Add a new API for the custom resource AppService
+$ operator-sdk add api --api-version=app.example.com/v1alpha1 --kind=AppService
```
Comment (LiliC) [inline README.md]: I am not sure why there are now 3 commands when you could use only one before. Was this in high demand so people just generate what they need. If so why not use flags instead of adding a subcommand. Just curious about the reasons behind this.

Comment (shawn-hurley) [reply README.md]: This was wanted for ansible operator. When an ansible operator wants to watch a new resource it only needs a CRD. Making an option to short circuit the API command and make it more clear to our users how to add the resources they will need was important to us. 

This was already implemented in the generate command and was moved to `add` as it is a better fit. I think that if we want to remove this option can we open an issue and/or set up a discussion?

Comment (LiliC) [reply README.md]: Oh, yes was just a question. Thanks!

File: commands/operator-sdk/cmd/new.go [hunk 20]
```diff
@@ -80,59 +83,162 @@ const (
...
-	if err != nil {
-		cmdError.ExitWithError(cmdError.ExitError, fmt.Errorf("failed to create project %v: %v", projectName, err))
-	}
-	if operatorType == goOperatorType {
+	switch operatorType {
+	case goOperatorType:
+		doScaffold()
 		pullDep()
-		generate.K8sCodegen(projectName)
+	case ansibleOperatorType:
```
Comment (joelanford) [inline commands/operator-sdk/cmd/new.go]: Should we normalize the constants for the operator types? We also have this:

https://github.com/operator-framework/operator-sdk/blob/refactor/rebase-master-add-AO-commands/commands/operator-sdk/cmd/cmdutil/util.go#L31-L36

Comment (shawn-hurley) [reply commands/operator-sdk/cmd/new.go]: So I think one is for the user(command line input) and one is for internal use when determining if ansible operator vs golang operator during build/up local.

If we think it is really important that these are the same then we need to choose the string versions and we lose the default value functionality of an iota.

Comment (shawn-hurley) [reply commands/operator-sdk/cmd/new.go]: So I think one is for the user(command line input) and one is for internal use when determining if ansible operator vs golang operator during build/up local.

If we think it is really important that these are the same then we need to choose the string versions and we lose the default value functionality of an iota.

File: pkg/scaffold/gitignore.go (hunk 11 above)
Comment (shawn-hurley) [reply pkg/scaffold/gitignore.go]: I am sorry, I don't understand the changes that I made to make these be removed, can you help me make sure that I didn't change something I should not have?

File: commands/operator-sdk/cmd/new.go (hunk 20 above)
Comment (joelanford) [reply commands/operator-sdk/cmd/new.go]: It's probably not important for this PR regardless, but something I noticed. Maybe another option is to add a helper/validation function that verifies that the string input is a valid operator type and returns the internal int-based const?

File: Makefile (hunk 15 above)
Comment (abhinavdahiya) [reply Makefile]: There is `--always` flag in `git describe`
https://git-scm.com/docs/git-describe#git-describe---always

**Comment by hasbro17** [comment]
> dep ensure as part of the new command, it always gives me the following warning dep: WARNING: redundant prune option "unused-packages" set for "k8s.io/code-generator"

@LiliC Previously we were pruning all unused pkgs by default in Gopkg.toml. But I've removed that since new dependencies for the controller-runtime are added when you run `operator sdk add api` or `add controller`.  So rather running dep ensure again after each cmd I just disabled the pruning of unused pkgs.
The warning is due to the redundant `unused-package=false` option for code-generator. That's not needed since we're not pruning anymore.
https://github.com/operator-framework/operator-sdk/blob/refactor/rebase-master-add-AO-commands/pkg/scaffold/gopkgtoml.go#L87

File: commands/operator-sdk/cmd/add/crd.go [hunk 21]
```diff
@@ -59,49 +56,64 @@ Generated CR  filename: <project-name>/deploy/<group>_<version>_<kind>_cr.yaml
 }
 
 func crdFunc(cmd *cobra.Command, args []string) {
+	cfg := &input.Config{
+		Repo:           cmdutil.MustGetwd(),
```
Comment (hasbro17) [inline commands/operator-sdk/cmd/add/crd.go]: @shawn-hurley @estroz 
I think Config.Repo is supposed to be the go import for the project and not the absolute path.
```Go
			cfg := &input.Config{
				Repo:           cmdutil.CheckAndGetCurrPkg(),
				AbsProjectPath: cmdutil.MustGetwd(),
			}
```
https://github.com/operator-framework/operator-sdk/blob/refactor/rebase-master-add-AO-commands/pkg/scaffold/input/input.go#L113-L114

Since the `Repo` field isn't really used in the CRD template this doesn't cause an issue but it really should be the project path relative to`$GOPATH/src`.

File: test/e2e/memcached_test.go (hunk 17 above)
Comment (estroz) [reply test/e2e/memcached_test.go]: After reverting this and merging, I can write a quick PR to update any branch with the branch annotation I added.

File: commands/operator-sdk/cmd/add/crd.go (hunk 21 above)
Comment (estroz) [reply commands/operator-sdk/cmd/add/crd.go]: Yes that is correct and should be changed to match your example.

Comment (shawn-hurley) [reply commands/operator-sdk/cmd/add/crd.go]: So an ansible user cannot create new CRDs unless they set up their project with a gopath? I don't think this is a change we want.

File: pkg/scaffold/gitignore_test.go [hunk 22]
```diff
@@ -0,0 +1,118 @@
...
+# temporary
+.netrwhist
+# auto-generated tag files
+tags
+### VisualStudioCode ###
+.vscode/*
+!.vscode/settings.json
+!.vscode/tasks.json
+!.vscode/launch.json
+!.vscode/extensions.json
```
Comment (hasbro17) [inline pkg/scaffold/gitignore_test.go]: @shawn-hurley Nit but we need to update the unit test and remove these as well.

File: commands/operator-sdk/cmd/new.go [hunk 23]
```diff
@@ -80,59 +83,162 @@ const (
...
+		&scaffold.GopkgToml{},
+	)
+	if err != nil {
+		log.Fatalf("new scaffold failed: (%v)", err)
+	}
+}
+
+func doAnsibleScaffold() {
+	cfg := &input.Config{
+		Repo:           filepath.Join(cmdutil.MustGetwd(), projectName),
```
Comment (estroz) [inline commands/operator-sdk/cmd/new.go]: `Repo` isn't used by Ansible scaffolds and should be removed.

File: commands/operator-sdk/cmd/add/crd.go (hunk 21 above)
Comment (estroz) [reply commands/operator-sdk/cmd/add/crd.go]: The `Repo` field isn't used by CRD or CR scaffolds, so it either shouldn't be set at all or should be set to the Go package location.

Comment (hasbro17) [reply commands/operator-sdk/cmd/add/crd.go]: No they still should be able to set it up.
`Repo ` `AbsProjectPath ` `ProjectName` are just fields common to all inputs.

For generating a CRD you only need `AbsProjectPath` which is set correctly. 
`Repo` is mostly used for Go files templates to set the go imports correctly.

You can just ignore `Repo` not set it.
```Go
			cfg := &input.Config{
				AbsProjectPath: cmdutil.MustGetwd(),
			}
```

File: Makefile (hunk 15 above)
Comment (estroz) [reply Makefile]: @shawn-hurley `git describe --dirty --tags --always`

File: commands/operator-sdk/cmd/add/crd.go (hunk 21 above)
Comment (shawn-hurley) [reply commands/operator-sdk/cmd/add/crd.go]: > The Repo field isn't used by CRD or CR scaffolds, so it either shouldn't be set at all or should be set to the Go package location.

We should probably rename this then for the future. `Repo` sounds like it is just the base repository directory, and I would assume that we would be able to use it in the future if we needed it. If you really think it should only be the  go package location then we may want to create a new field for that. As @hasbro17 said though we can just remove them right now from our new scaffolds and we can come back to this.

File: commands/operator-sdk/cmd/new.go [hunk 24]
```diff
@@ -80,59 +83,161 @@ const (
...
+		if err != nil {
+			log.Fatalf("new scaffold failed: (%v)", err)
+		}
+	}
+
+	// Run galaxy init.
+	cmd := exec.Command(filepath.Join(galaxyInit.AbsProjectPath, galaxyInit.Path))
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.Run()
```
Comment (hasbro17) [inline commands/operator-sdk/cmd/new.go]: Check return error and exit?

Or do we still need to remove the galaxy init script below?

File: commands/operator-sdk/cmd/new.go [hunk 25]
```diff
@@ -235,3 +306,66 @@ func initGit() {
 	execCmd(os.Stdout, "git", "commit", "-q", "-m", "INITIAL COMMIT")
 	fmt.Fprintln(os.Stdout, "Run git init done")
 }
+
+// Copied from add/api.go command
+func updateRoleForResource(r *scaffold.Resource, absProjectPath string) error {
```
Comment (estroz) [inline commands/operator-sdk/cmd/new.go]: If this is an exact copy of `updateRoleForResource`, I think this should live in `cmd/util` unless you have a reason to have another copy.

File: commands/operator-sdk/cmd/new.go [hunk 26]
```diff
@@ -143,50 +248,48 @@ func repoPath() string {
...
-	}
-	kindFirstLetter := string(kind[0])
-	if kindFirstLetter != strings.ToUpper(kindFirstLetter) {
-		cmdError.ExitWithError(cmdError.ExitBadArgs, errors.New("--kind must start with an uppercase letter"))
+		log.Fatal("--generate-playbook can only be used with --type `ansible`")
 	}
-	if strings.Count(apiVersion, "/") != 1 {
-		cmdError.ExitWithError(cmdError.ExitBadArgs, fmt.Errorf("api-version has wrong format (%v); format must be $GROUP_NAME/$VERSION (e.g app.example.com/v1alpha1)", apiVersion))
+	if operatorType == goOperatorType && (len(apiVersion) != 0 || len(kind) != 0) {
+		log.Fatal("go type operator does not use --api-version or --kind. Please see generate command after running new.")
```
Comment (hasbro17) [inline commands/operator-sdk/cmd/new.go]: Nit:
Change `Please see generate command after running new`
to `Please see "operator-sdk add" command after running new`

File: commands/operator-sdk/cmd/new.go (hunk 25 above)
Comment (shawn-hurley) [reply commands/operator-sdk/cmd/new.go]: It is an exact copy. I think that we can refactor in a bit.

File: commands/operator-sdk/cmd/new.go (hunk 24 above)
Comment (shawn-hurley) [reply commands/operator-sdk/cmd/new.go]: I think we need to remove the galaxy init script. If the command errors the right info will already be printed to the console.

File: commands/operator-sdk/cmd/add/api.go [hunk 27]
```diff
@@ -146,7 +149,13 @@ func updateRoleForResource(r *scaffold.Resource, absProjectPath string) error {
 			role.Rules = append(role.Rules, *pr)
 		}
 		// update role.yaml
-		data, err := yaml.Marshal(&role)
+		d, err := json.Marshal(&role)
```
Comment (estroz) [inline commands/operator-sdk/cmd/add/api.go]: Was `yaml.Marshal` not returning the expected yaml for a role? Why is this change necessary?

File: commands/operator-sdk/cmd/new.go (hunk 25 above)
Comment (hasbro17) [reply commands/operator-sdk/cmd/new.go]: We can add a ToDo for now to consolidate them later.

File: commands/operator-sdk/cmd/add/api.go (hunk 27 above)
Comment (shawn-hurley) [reply commands/operator-sdk/cmd/add/api.go]: Yeah, that was not working at all. 

yaml.Marshal was not outputting the correct yaml, this is because there are no tags for yaml in the role struct, meaning 1. names were incorrect and 2. that type meta was being printed.

File: pkg/scaffold/ansible/galaxy_init.go [hunk 28]
```diff
@@ -0,0 +1,58 @@
...
+
+const galaxyInitTmpl = `#!/usr/bin/env bash
+
+if ! which ansible-galaxy > /dev/null; then
+	echo "ansible needs to be installed"
+	exit 1
+fi
+
+echo "Initializing role skeleton..."
+ansible-galaxy init --init-path={{.Input.AbsProjectPath}}/roles/ {{.Resource.Kind}}
```
Comment (estroz) [inline pkg/scaffold/ansible/galaxy_init.go]: Nit: the `.Input` parent isn't necessary since `input.Input` is embedded in `GalaxyInit`.

File: pkg/scaffold/ansible/galaxy_init.go (hunk 6 above)
Comment (estroz) [reply pkg/scaffold/ansible/galaxy_init.go]: This should be removed if you're setting `Dir` in the caller. I'm not sure why `Dir` is necessary since its not a template pipeline and not used anywhere as far as I can tell.
